runtime/plaid/src/apis/aws/dynamodb.rs

```diff
@@ -0,0 +1,538 @@
+use aws_sdk_dynamodb::primitives::Blob;
+use aws_sdk_dynamodb::types::{AttributeValue, ReturnValue};
+use aws_sdk_dynamodb::Client;
+use serde_json::{json, Map, Value};
+
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
+
+use serde::Deserialize;
+
+use crate::{get_aws_sdk_config, loader::PlaidModule, AwsAuthentication};
+
+/// Defines configuration for the KMS API
+#[derive(Deserialize)]
+pub struct DynamoDbConfig {
+    /// This can either be:
+    /// - `IAM`: Uses the IAM role assigned to the instance or environment.
+    /// - `ApiKey`: Uses explicit credentials, including an access key ID, secret access key, and region.
+    authentication: AwsAuthentication,
+    /// Configured writers - maps a table name to a list of rules that are allowed to write it
+    write: HashMap<String, HashSet<String>>,
+    /// Configured readers - maps a table name to a list of rules that are allowed to read it
+    read: HashMap<String, HashSet<String>>,
+}
+
+/// Represents the KMS API that handles all requests to KMS
+pub struct DynamoDb {
```

Comment:
Add a comment here that if Plaid is configured with the DynamoDB database backend, sharing tables here will lead to undefined behaviour.